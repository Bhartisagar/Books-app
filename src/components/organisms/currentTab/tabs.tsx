import React,{useState,useEffect} from 'react'
import {Tabs,Tab,Container,Grid, makeStyles } from '@material-ui/core'
import Cards from '../../molecules/BookCard';
import api from '../../../api';


export const TabPanel=(props:any)=>
{ const {children,value,index} = props;

return  (
  <div>
     {value===index && <h1 > {children} </h1> }  

</div>

)
}
interface DataBook{
   
  "id": number,
  "imgSrc": string,
  "imgAlt": string,
  "name": string,
  "author": string,
  "readTime": string,
  "readCount": string,
  "state": string,
  "category": string,
  "audio": boolean,
  "synopsis"?:string,
"justAdded": boolean,
"isTrending": boolean

}

export const MyTabs=()=>{
  
const [crcol,setcrColor] =useState('black');
const [rcol,setrColor] =useState('black');
const [bookState,setbookState]=useState(false)

const useStyles=makeStyles({
  tabStyle:{
    margin:'10px 280px',
},
cr:{
  color:crcol,
  borderBottomColor:crcol,
  BorderBottom:'3px',
  borderBottomStyle:'solid',
  width:"346px",

},
r:{
  color:rcol,
  width:"346px",
}

})
const setState=(data:DataBook)=>{
  if(data.state==='CR'){
    data.state='FR'
    setbookState(!bookState)
    console.log(data)
 }
 else{
    data.state='CR'
    setbookState(!bookState)
 }
 api.put(`/${data.id}`,data).then(res=>res.data).then((status)=>{console.log(status)})


}
  const [bookData,setBookData] = useState([]);
  

    useEffect(()=>{
      api.get('/')
      .then(response => {
      setBookData(response.data);
        }, error => {
         console.log(error);
      });
    },[bookState])
    const [value,setValue] =useState(0);

    const handleTabs=(val):void=>{
        setValue(val)
        if(val===0){
          setcrColor('#2CE080');
          setrColor('black');

        }
        else{
          setrColor('#2CE080');
          setcrColor('black')
        }

    }
    
    const classes=useStyles();
  return (
    <div>
            <Tabs value={value}  data-testid={'tabs'} onChange={handleTabs} className={classes.tabStyle} TabIndicatorProps={{
style: {
  backgroundColor: "rgba(44, 224, 128, 1)"
 }
}}>
                <Tab label="Current Reading" id='curTab' className={classes.cr}></Tab>
                <Tab label="finished" id='finishTab' className={classes.r} ></Tab>             
            </Tabs>
           <TabPanel  index={0} value={value} >
           <Container>
        <Grid container >
        {bookData && bookData.filter((book:DataBook)=>{return book.state==='CR';}).map((data:DataBook)=>(
          <Grid style={{padding:'10px 0 10px 15px'}} key={data.id} data-testid={`current-${data.id}`}>
          <Cards bookData={data} btnText={"Finished"} bookFunction={(dataBook:DataBook)=>{setState(dataBook)}} />
            
         </Grid>
         )
        )
       }

        </Grid>
         </Container>
           </TabPanel>

           <TabPanel  index={1} value={value}>
            <Container>
        <Grid container >
         {bookData && bookData.filter((book:DataBook)=>{
           return book.state==='FR';
         }).map((data:DataBook)=>(
           <Grid style={{padding:'10px 0 10px 15px'}} key={data.id} data-testid={`finished-${data.id}`}>
           <Cards bookData={data} btnText={"Read Again"} bookFunction={(finishedBook:DataBook)=>{setState(finishedBook)}}/>
             
          </Grid>
          )
         )
        }

        </Grid>
         </Container>
          </TabPanel>
        
    </div>
  )
}
