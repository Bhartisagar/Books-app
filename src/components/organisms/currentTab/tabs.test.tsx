import { render, screen } from "@testing-library/react";
import { MyTabs } from "./tabs";
import renderer from 'react-test-renderer';
import { fireEvent } from "@storybook/testing-library";

test("first snapshot",()=>{
    const component = renderer.create(
        <MyTabs></MyTabs>
    )
    let tree=component.toJSON();

    expect(tree).toMatchSnapshot();
})

test('should render tab Current tab',()=>{
    render(<MyTabs></MyTabs>)
    const name = screen.getByText("Current Reading")
    expect(name).toBeInTheDocument();

})
test('should render tab finished',()=>{
    render(<MyTabs></MyTabs>)
    const name = screen.getByText("finished")
    expect(name).toBeInTheDocument();

})
test('should render tab',()=>{
    render(<MyTabs></MyTabs>)
    const name = screen.getByRole("tab",{selected:true})
    expect(name).toBeInTheDocument();

})
test('should render mapp data button',async()=>{
    render(<MyTabs></MyTabs>)
    const card = await screen.findByTestId("current-3")
    expect(card).toBeInTheDocument();
})

test('should render mapp data',async()=>{
    render(<MyTabs></MyTabs>)
    const card = await screen.findByTestId("current-8")
    expect(card).toBeInTheDocument();
})

test('should render mapp dat',async()=>{
    render(<MyTabs></MyTabs>)
    const card = await screen.findByTestId("cards1")
    expect(card).toBeInTheDocument();
})

test('should render mapp databutton',async()=>{
    render(<MyTabs></MyTabs>)
    const card1 = await screen.findByTestId("button-1")
    expect(card1).toBeInTheDocument();
})

test('should render mapp data headings',async()=>{
    render(<MyTabs></MyTabs>)
    const card2 = await screen.findByTestId("headings-1")
    expect(card2).toBeInTheDocument();
})
test('should render  onchange',async()=>{
    render(<MyTabs></MyTabs>)
    const card2 = screen.getByTestId('tabs')
    fireEvent.change(card2)
})
test('should render mapp databutton fun',async()=>{
    render(<MyTabs></MyTabs>)
    const card1 = await screen.findByTestId("button-1")
   fireEvent.click(card1)
})
