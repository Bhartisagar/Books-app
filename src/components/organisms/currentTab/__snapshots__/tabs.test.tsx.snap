// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot 1`] = `
<div>
  <div
    className="MuiTabs-root makeStyles-tabStyle-1"
    data-testid="tabs"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-cr-2 Mui-selected"
          disabled={false}
          id="curTab"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Current Reading
          </span>
          <span
            className="PrivateTabIndicator-root-4 PrivateTabIndicator-colorSecondary-6 MuiTabs-indicator"
            style={
              Object {
                "backgroundColor": "rgba(44, 224, 128, 1)",
              }
            }
          />
        </button>
        <button
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit makeStyles-r-3"
          disabled={false}
          id="finishTab"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            finished
          </span>
        </button>
      </div>
    </div>
  </div>
  <div>
    <h1>
       
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          className="MuiGrid-root MuiGrid-container"
        />
      </div>
       
    </h1>
  </div>
  <div />
</div>
`;
